homeassistant/components/foscam/camera.py

```diff
@@ -114,6 +125,31 @@ async def async_handle_ptz(service):
         FOSCAM_DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
     )
 
+    async def async_handle_ptz_preset(service):
+        """Handle PTZ preset service call."""
+        preset_name = service.data[ATTR_PRESET_NAME]
+        entity_ids = await async_extract_entity_ids(hass, service)
+
+        if not entity_ids:
+            return
+
+        _LOGGER.debug("Moving to preset '%s' camera(s): %s", preset_name, entity_ids)
+
+        all_cameras = hass.data[FOSCAM_DATA][FOSCAM_ENTITIES]
+        target_cameras = [
+            camera for camera in all_cameras if camera.entity_id in entity_ids
+        ]
+
+        for camera in target_cameras:
+            await camera.async_perform_ptz_preset(preset_name)
+
+    hass.services.async_register(
+        FOSCAM_DOMAIN,
+        SERVICE_PTZ_PRESET,
+        async_handle_ptz_preset,
+        schema=SERVICE_PTZ_PRESET_SCHEMA,
+    )
```

Comment:
This should be adjusted to use an entity service:

<https://developers.home-assistant.io/docs/dev_101_services#entity-services>

Reply:
I rebased and rewrote the code to use an entity service.  Please let me know if it looks good.

Reply:
Please don't use squash & force push in open source projects with reviews, it makes it very hard for the reviewer to see what you have changed.

We will squash it on merge 😉

Reply:
Sorry, I thought I was doing the right thing by looking at what was done in #41429 and [this](https://developers.home-assistant.io/docs/development_catching_up).  Can you tell me what the proper procedure would have been?  Since there were changes to the dev branch after my original commit in November.

Reply:
Rebase and make separate commits for the other changes you've made.
But I can see the possible trouble it could have caused as well, so 👍